src/cpu/flexcpu/simple_dataflow_cpu.hh

```diff
@@ -49,6 +49,7 @@ class SDCPUThread;
  */
 class SimpleDataflowCPU : public BaseCPU
 {
+    friend class SDCPUThread;
```

Comment:
:/ Rather than doing this, why not pass the parameter through to the thread?

This is creating that coupling between the CPU and thread that you didn't want :).

Reply:
I'll make this change tomorrow, I'm working on the miscreg thing right now

Reply:
I guess one thing I'm scared of is just the constructor exploding in size if this is done a lot...

Reply:
It shouldn't... If we end up with many more things that need to be passed through we can revisit this.

Reply:
I think I'll just get rid of the friend class, but either expose or add accessors to the stored parameter

Reply:
but on the other hand, serialization is really the thread's job, not the CPU's :l